### PR TITLE
Make drinks placeable in the world

### PR DESCRIPTION
**Pull request by todada**
- Changes craftitems to node definitions for glasses and bottles
- Adds support for vessels shelf

### CODE REVIEW

**Review by NathanSalapat — COMMENTED** [review]
Changes look good. Will try this out when I get back to my computer. I see you also added the steel bottle, nice work.

File: drink_machines.lua [hunk 1]
```diff
@@ -397,6 +417,9 @@ minetest.register_node('drinks:liquid_barrel', {
       if outputstack == 'vessels:glass_bottle' then
          drinks.drinks_liquid_sub(4, 'jbo', 128, pos)
       end
+      if outputstack == 'vessels:steel_bottle' then
```
Comment (sofar) [inline drink_machines.lua:420]: wow, someone needs to really:

1) learn `elseif` and why it matters
2) migrate this to a lookup table!

Comment (NathanSalapat) [reply drink_machines.lua:420]: elseif is clearly my fault, I don't use those much, but I can change that, especially if it improves performance. I'll have to learn about lookup tables, as I've never heard of those, but I think I should be able to implement that as well.

Comment (sofar) [reply drink_machines.lua:420]: so with elseif's you exit the entire list once a hit has been found. Now each of the conditions is always run.

The whole section can be replaced with a simple table lookup:

(untested, concept code :))
```
local tbl = {
   ["vessels:steel_bottle"] = {name = "jsb", size = 256},
   ["vessels:...."]  = {name = "......

...

drinks.dinks_liquid_sub(4, tbl[outputstack].name, tbl[outputstack].size, pos)
```

Comment (todada) [reply drink_machines.lua:420]: Well, those if-chains just happen when something starts small and grows more and more. 
I didn't wanted to change the structure too much, but adding the 5th condition I agree it's somehow time for a table ;-).
I would suggest also to combine drinks.lua and drink2.lua, as they share mostly the same content.

Comment (NathanSalapat) [reply drink_machines.lua:420]: Drinks and drinks two are almost the exact same, but one has hydration values for use with the thirst mod. I'll see if I can't somehow make that still work with only one file.

**Pull request merged**